File: Data Structures & Algorithms/find-duplicate-integer/submission-1.py
class Solution:
    def findDuplicate(self, nums: List[int]) -> int:
        fast = 1
        slow = 0
        size = len(nums)
        while (nums[slow] != nums[fast]):
            slow = (slow + 1)%size
            fast = (fast + 2 )%size
            if slow == fast:
                fast = (fast + 1) %size
    

        return nums[slow]

File: Data Structures & Algorithms/top-k-elements-in-list/submission-1.java
class Solution {
    public int[] topKFrequent(int[] nums, int k) {
        Map<Integer, Integer> counter = new HashMap<>();
        List<Integer>[] freq = new List[nums.length + 1];

        for (int i=0; i<freq.length; i++) {
            freq[i] = new ArrayList<>();
        }

        for (int num: nums) {
            counter.put(num, counter.getOrDefault(num, 0) + 1);
        }

        for (Map.Entry<Integer, Integer> entry: counter.entrySet()) {
            int val = entry.getKey();
            int frequency = entry.getValue();
            freq[frequency].add(val);
        }
        
       int[] res = new int[k];
        int index = 0;
        for (int i=freq.length-1; i>=0; i--) {
            for (int val: freq[i]) {
                if (index == k) {
                    return res;
                }
                res[index]=val;
                index++;
            }
        }
        return res;
    }
}
